File: GotDotNet.com/ES Hands On Lab/VB/Exercise 2/HelloEnterpriseServices/HelloTx.vb
Imports System.Windows.Forms
'
' TODO: Add import and reference for EnterpriseServices
'


' 
' TODO: Add Transaction attribute to the HelloTx class
' TODO: HelloTx class must inherit from ServicedComponent
'
Public Class HelloTx

    Public Sub SayHello(ByVal Name As String)
        '
        ' TODO: Get the transaction ID from the ContextUtil object
        '

        '
        ' TODO: Ask the user if they want to commit the transaction
        '
        MessageBox.Show(String.Format("Hello {0}", Name))

        '
        ' TODO: If the user said Yes, commit, otherwise abort
        '
    End Sub

End Class

File: GotDotNet.com/ES Hands On Lab/CS/Exercise 2/HelloEnterpriseServices/HelloTx.cs
using System;
using System.Windows.Forms;
//TODO: Add using System.EnterpriseServices;

namespace HelloEnterpriseServices
{
	//
	// TODO: Add a Transaction attribute to the HelloTx class
	// TODO: Derive HelloTx from ServicedComponent
	//
	public class HelloTx
	{
		public HelloTx()
		{
		}

		public void SayHello(string Name)
		{
			//
			// TODO: Get the transaction ID
			//

			//
			// TODO: Ask the user if they want to commit the transaction
			//
			MessageBox.Show(string.Format("Hello {0}", Name));

			// 
			// TODO: Call ContextUtil SetComplete or SetAbort
			//
		}
	}
}
